integration/test/Testlib/MockIntegrationService.hs

```diff
@@ -107,30 +109,43 @@ withMockServer mkApp go = withFreePortAnyAddr $ \(sPort, sock) -> do
   let tlss = Warp.tlsSettingsMemory (cs mockServerCert) (cs mockServerPrivKey)
   let defs = Warp.defaultSettings {Warp.settingsPort = sPort, Warp.settingsBeforeMainLoop = putMVar serverStarted ()}
   buf <- newChan
-  srv <- async . liftIO . Warp.runTLSSocket tlss defs sock $ mkApp buf
+  srv <- async $ withRunInIO \inIO -> do
+    Warp.runTLSSocket tlss defs sock \req respond -> do
+      inIO $ mkApp buf req (liftIO . respond)
   srvMVar <- UnliftIO.Timeout.timeout 5_000_000 (takeMVar serverStarted)
   case srvMVar of
     Just () -> go sPort buf `finally` cancel srv
     Nothing -> error . show =<< poll srv
 
+lhMockApp :: Chan (Wai.Request, LBS.ByteString) -> LiftedApplication
+lhMockApp = lhMockApp' Nothing
+
 -- | LegalHold service.  Just fake the API, do not maintain any internal state.
-lhMockApp :: Chan (Wai.Request, LBS.ByteString) -> Wai.Application
-lhMockApp ch req cont = do
+lhMockApp' :: Maybe (Value, [Value]) -> Chan (Wai.Request, LBS.ByteString) -> LiftedApplication
+lhMockApp' mks ch req cont = withRunInIO \inIO -> do
```

Comment:
```suggestion
lhMockAppWithPrekeys mks ch req cont = withRunInIO \inIO -> do
```

or even better: use the patter as for `createUser`, `CreateUser`, `instance Default CreateUser`.